agent/connect/sni.go

```diff
@@ -65,6 +65,39 @@ func ServiceSNI(service string, subset string, namespace string, partition strin
 	}
 }
 
+func dotSplitLast(s string, n int) string {
+	tokens := strings.SplitN(s, ".", n)
+	if len(tokens) != n {
+		return ""
+	}
+	return tokens[n-1]
+}
+
+func TrustDomainForTarget(target structs.DiscoveryTarget) string {
+	if target.External {
+		return ""
+	}
+
+	switch target.Partition {
+	case "default":
+		if target.ServiceSubset == "" {
+			// service, namespace, datacenter, internal, trustDomain
+			return dotSplitLast(target.SNI, 5)
+		} else {
+			// subset, service, namespace, datacenter, internal, trustDomain
+			return dotSplitLast(target.SNI, 6)
+		}
+	default:
+		if target.ServiceSubset == "" {
+			// service, namespace, partition, datacenter, internalVersion, trustDomain
+			return dotSplitLast(target.SNI, 6)
+		} else {
+			// subset, service, namespace, partition, datacenter, internalVersion, trustDomain
+			return dotSplitLast(target.SNI, 7)
```

Comment:
where does the `5` `6` and `7` numbers come from? should we define those as constants?